node-tests/blueprints/service-test.js

```diff
@@ -131,4 +178,41 @@ describe('Blueprint: service', function() {
       });
     });
   });
+
+  describe('in addon - module unification', function() {
+    beforeEach(function() {
+      return emberNew({ target: 'addon' }).then(() => fs.ensureDirSync('src'));
+    });
+
+    it('service foo', function() {
+      return emberGenerateDestroy(['service', 'foo'], _file => {
+        expect(_file('src/services/foo.js'))
+          .to.contain("import Service from '@ember/service';")
+          .to.contain('export default Service.extend({\n});');
+
+        //TODO: GJ: how does this get generated?
+        expect(_file('app/services/foo.js'))
+          .to.not.exist;
```

Comment:
we'll need to update https://github.com/ember-cli/ember-cli/tree/master/blueprints/addon-import to be MU aware

Reply:
even simpler: implement `supportsAddon` in the blueprint

Reply:
Hmm, `supportsAddon` doesn't currently have access to `project`. Another option, modify the following to exclude `project.isModuleUnification`:

https://github.com/ember-cli/ember-cli/blob/758d269cc7f885183eac3aab948b8561574b19bd/lib/tasks/generate-from-blueprint.js#L26

Reply:
https://github.com/ember-cli/ember-cli/pull/7702

Reply:
This PR has landed